packages/kbn-es/src/utils/read_roles_from_resource.ts

```diff
@@ -20,3 +20,15 @@ export const readRolesFromResource = (resourcePath: string) => {
     throw new Error(`expected ${resourcePath} file to parse to an object`);
   }
 };
+
+export const readRolesDescriptorsFromResource = (resourcePath: string) => {
+  if (!fs.existsSync(resourcePath) || extname(resourcePath) !== '.yml') {
+    throw new Error(`${resourcePath} does not exist or not a yml file`);
+  }
+  const data = loadYaml(fs.readFileSync(resourcePath, 'utf8'));
+  if (typeof data === 'object' && data !== null) {
+    return data;
+  } else {
+    throw new Error(`expected ${resourcePath} file to parse to an object`);
+  }
+};
```

Comment:
We probably can get rid of `readRolesFromResource` (only 1 usage) and just use `readRolesDescriptorsFromResource` for all cases?